internal/bminventory/inventory.go

```diff
@@ -5066,6 +5061,8 @@ func (b *bareMetalInventory) V2UpdateHostIgnitionInternal(ctx context.Context, p
 			log.WithError(err).Errorf("Failed to parse host ignition config patch %s", params.HostIgnitionParams)
 			return nil, common.NewApiError(http.StatusBadRequest, err)
 		}
+	} else {
+		log.Infof("Applying empty host ignition config to host %s in infra-env %s", params.HostID, params.InfraEnvID)
```

Comment:
Meaning of this path is that providing non-empty `params.HostIgnitionParams` and empty `params.HostIgnitionParams.Config` is equivalent to saying "I want to remove the custom ignition override". Just for the clarity when debugging potential issues, I'm adding a log line here so that it's obvious post-factum